frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -225,6 +225,15 @@ func Dockerfile2LLB(ctx context.Context, dt []byte, opt ConvertOpt) (*llb.State,
 			}
 			func(i int, d *dispatchState) {
 				eg.Go(func() error {
+					name, err := shlex.ProcessWordWithMap(d.stage.BaseName, metaArgsToMap(optMetaArgs))
```

Comment:
This one is more a proof of concept but a final solution. I'm going to find a better place for expanding the `BaseName` and maybe we don't even have to make it twice (see `(c *CopyCommand) Expand(...)`).